Controller/Adminhtml/Item/MassDelete.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace SomethingDigital\AuctionItem\Controller\Adminhtml\Item;
+
+use Magento\Backend\App\Action;
+use Magento\Backend\App\Action\Context;
+use Magento\Backend\Model\View\Result\Redirect;
+use Magento\Framework\App\Action\HttpPostActionInterface;
+use Magento\Framework\Controller\ResultFactory;
+use Magento\Framework\Exception\NotFoundException;
+use Magento\Ui\Component\MassAction\Filter;
+use SomethingDigital\AuctionItem\Model\AuctionItemRepository;
+use SomethingDigital\AuctionItem\Model\ResourceModel\AuctionItem\CollectionFactory;
+
+/**
+ * Custom Controller for AuctionItem MassDelete
+ *
+ * Class MassDelete
+ */
+class MassDelete extends Action implements HttpPostActionInterface
+{
+    /**
+     * Authorization level
+     */
+    const ADMIN_RESOURCE = 'SomethingDigital_AuctionItem::grid';
+
+    /**
+     * @var CollectionFactory
+     */
+    protected $collectionFactory;
+
+    /**
+     * @var AuctionItemRepository
+     */
+    private $auctionItemRepository;
+
+    /**
+     * @var Filter
+     */
+    protected $filter;
+
+    /**
+     * Constructor
+     *
+     * @param Context $context
+     * @param Filter $filter
+     * @param CollectionFactory $collectionFactory
+     * @param AuctionItemRepository $auctionItemRepository
+     */
+    public function __construct(
+        Context $context,
+        Filter $filter,
+        CollectionFactory $collectionFactory,
+        AuctionItemRepository $auctionItemRepository
+    ) {
+        $this->filter = $filter;
+        $this->collectionFactory = $collectionFactory;
+        $this->auctionItemRepository = $auctionItemRepository;
+        parent::__construct($context);
+    }
+
+    /**
+     * AuctionItem delete action
+     *
+     * @return Redirect
+     */
+    public function execute(): Redirect
+    {
+        if (!$this->getRequest()->isPost()) {
+            throw new NotFoundException(__('Page not found'));
+        }
+        $collection = $this->filter->getCollection($this->collectionFactory->create());
+        $auctionItemsDeleted = 0;
+        foreach ($collection->getItems() as $auctionItem) {
+            $this->auctionItemRepository->delete($auctionItem);
```

Comment:
My initial reaction to seeing this was that it would be better to bulk delete the auction items through the Collection. This way we could perform the mass delete by making only one trip to the DB instead of one trip for each item. I looked around a bit, but I couldn't find anything really.
It seems like this is how it's handled in certain places of core Magento: https://magento.stackexchange.com/a/118144

So I think it's totally fine to do the mass delete like this, but I wanted to share my thinking anyway